src/lib/dma.c

```diff
@@ -128,7 +128,10 @@ void dma_put(struct dma *dma)
 	k_spinlock_key_t key;
 
 	key = k_spin_lock(&dma->lock);
-	--dma->sref;
+	if (--dma->sref == 0) {
+		rfree(dma->chan);
+		dma->chan = NULL;
+	}
```

Comment:
Without this, we lose one "dma->chan = rzalloc(SOF_MEM_ZONE_RUNTIME_SHARED" worth of memory whenever sref goes to 1->0 and back.